File: backend/orm/message.py
from sqlalchemy.orm import declarative_base, relationship
from sqlalchemy import Column, TEXT, DateTime, ForeignKey
from fastapi_utils.guid_type import GUID
from datetime import datetime


Base = declarative_base()


class Message(Base):
    __tablename__ = "message"

    id = Column(GUID, primary_key=True)
    content = Column(TEXT, nullable=False)
    created_at = Column(DateTime, nullable=False, default=datetime.now)
    update_at = Column(DateTime, nullable=False, default=datetime.now)

    user_id = Column(GUID, ForeignKey("user.id"))
    user = relationship("User", back_populates="messages")

    auction_user_id = Column(GUID, ForeignKey("auction_user.id"))
    auction_user = relationship("AuctionUser", back_populates="messages")

File: backend/src/models/message.js
const { DataTypes } = require('sequelize');

const sequelize = require('../../conf/sequelize');
const User = require('./user');


const Message = sequelize.define('message', {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    content: {
        type: DataTypes.TEXT,
        allowNull: false,
    },
},
    {
        tableName: 'message',
    }
);


Message.belongsTo(User, { foreignKey: 'user_id' })
User.hasMany(Message, { foreignKey: 'user_id' });

Message.belongsTo(User, { foreignKey: 'user_2_id' })
User.hasMany(Message, { foreignKey: 'user_2_id' });

module.exports = Message;
